python/pyspark/streaming/tests.py

```diff
@@ -1051,6 +1062,9 @@ def test_kafka_stream(self):
 
     def test_kafka_direct_stream(self):
         """Test the Python direct Kafka stream API."""
+        if not are_kafka_tests_enabled:
```

Comment:
Wouldn't it be simpler to follow the kinesis example?

The code checks this once in the `__main__` section, and if tests are disabled the suite is not added to the list of suites to be run.

Also, am I reading things wrong, or there are no tests for the 0.10 backend here? Is the 0.10 backend supported in pyspark?

Reply:
@vanzin ah I was distracted by how it's also handled in KinesisStreamTests. Done, I think.

I think that yes there is no 0.10 test, as I don't think Pyspark supports 0.10 (right @jerryshao ?): https://github.com/apache/spark/pull/14340#issuecomment-235457517

So all Kafka Pyspark tests should be disabled, yeah.
But, does this mean that Pyspark Kafka support is effectively deprecated, entirely?

(I need to add deprecation messages to 0.8 code)